### DISCUSSION TITLE
Can something be done about python pkages when python upgtrades ?

### DISCUSSION THREAD

**Discussion by shadmansaleh** [termux/termux-packages#7743] [Ideas]
Recently python package was upgraded to 3.10 . I had several packages installed with pip in py3.9 . Those packages are still in $PREFIX/lib/python-3.9/site-packages but now python-3.10 uses $PREFIX/lib/python-3.10/site-packages . So none of those are available on python-3.10 installation . Everytime python upgrades like this I need to manually clean up older site-packages folder and reinstall all the modules manualy .This is annoying. Is it possible for pkg manager to handle that ? If it could have a post upgrade hook or something so it checks what packages were installed in older version and install them on new version during upgrade . That will be cool . Anyone have similar issues ?

**Comment by ghost** [comment]
> This is annoying. Is it possible for pkg manager to handle that ?

For now it deletes modules only on complete deinstallation of Python package.

> checks what packages were installed in older version and install them on new version during upgrade

Python module installation has high chance of failure (especially if upgrading to the newest Python version), so automatic re-installation will not be implemented. At max it is possible to wipe them on version upgrade.

**Reply by shadmansaleh** [reply]
> Python module installation has high chance of failure (especially if upgrading to the newest Python version), so automatic re-installation will not be implemented. 

That makes sense.
> At max it is possible to wipe them on version upgrade.

Don't wipe them . I'll have no way of knowing what modules I had installed for reinstallation 😕 . And cleaning the old modules is the easy part of the process just a rm command.
